aws_xray_sdk/core/models/entity.py

```diff
@@ -12,21 +13,26 @@
 
 
 log = logging.getLogger(__name__)
+_valid_name_characters = string.ascii_letters + string.digits + '_.:/%&#=+\-@'
 
 
 class Entity(object):
     """
     The parent class for segment/subsegment. It holds common properties
     and methods on segment and subsegment.
     """
+
     def __init__(self, name):
 
         # required attributes
         self.id = self._generate_random_id()
-        self.name = name
+        self.name = ''.join([c for c in name if c in _valid_name_characters])
         self.start_time = time.time()
         self.parent_id = None
 
+        if self.name != name:
+            log.warning("Removing Name invalid characters.")
```

Comment:
Use segment/subsegment name instead of name could be clearer when user see such log entries.